cloud/amazon/ec2_vpc.py

```diff
@@ -209,8 +209,9 @@ def find_vpc(module, vpc_conn, vpc_id=None, cidr=None):
             # Get all tags for each of the found VPCs
             vpc_tags = dict((t.name, t.value) for t in vpc_conn.get_all_tags(filters={'resource-id': vpc.id}))
 
-            # If the supplied list of ID Tags match a subset of the VPC Tags, we found our VPC
-            if resource_tags and set(resource_tags.items()).issubset(set(vpc_tags.items())):
+            # If the supplied list of ID Tags match a subset of the VPC Tags,
+            # or if the supplied list of ID Tags match a superset of the VPC Tags, we found our VPC
+            if resource_tags and set(resource_tags.items()).issubset(set(vpc_tags.items())) or set(resource_tags.items().issuperset(set(vpc_tags.items()))):
```

Comment:
This is behavior changing as new VPCs could now be identified by existing plays that didn't earlier, and thus makes it backwards incompatible.  Mind you, I didn't add the original resource tag search for this module, so I don't really know all the use cases.  I think someone should take a look at this that does.
That said, it does seem a bit strange that searching for `["a", "b", "c", "d"]` would match VPCs with just `"a"`, but it could be useful.  It just seems a little risky, especially when combined with the existing subset behavior. 

Either way, it can slightly simplified as 
`if resource_tags and len(set(resource_tags.items().intersection(set(vpc_tags.items))) > 1`